Update to Rust 2024 edition and latest dependencies

n8behavior opened the pull request:
## Summary
Updates mdbook-rust-doc to work with modern Rust and adds function documentation support.

## Changes
- **Rust 2024 edition** (was 2018)
- **Updated dependencies**: syn 2.0, clap 4.5, and all others to latest versions
- **Added function support**: Implemented `Item::Fn` handling (was `bail!("Todo item type: Fn")`)
- **Fixed breaking API changes**: syn 2.0, clap 4.5, pulldown-cmark-to-cmark
- **Fixed doc comment parsing**: Properly trim leading spaces
- **Updated test configuration**: Correct paths and HTML output

## Problem
The preprocessor was incompatible with current Rust toolchains and couldn't extract documentation from standalone functions.

## Testing
- All unit tests passing
- Test book builds successfully  
- Tested extracting docs from functions (sync/async), structs, enums, impl blocks

## Compatibility
- Rust 1.89.0+
- mdBook 0.4.52
- Fully backward compatible

## Example
```rust
/// Initializes the application
pub async fn run() -> Result<()> { }
```

```markdown
{{ #rustdoc my_crate::run }}
```

n8behavior committed:
Update project to work with latest Rust dependencies (2025)

- Update all dependencies to latest versions in Cargo.toml
- Update syn to 2.0 with required features (full, parsing, extra-traits)
- Fix syn API changes:
  - ImplItem::Method -> ImplItem::Fn
  - attr.path -> attr.path() method call
  - Rewrite attribute parsing to use new meta API
  - Remove deprecated Item::Macro2 variant
- Update clap from v3 to v4.5:
  - Clap trait -> Parser trait
  - Add derive feature
  - Update derive macros
- Fix pulldown_cmark_to_cmark API (remove third parameter)
- Apply clippy suggestions for cleaner code
- Set edition to 2024
- Add .claude/settings.local.json to gitignore

All tests passing, project builds successfully with modern toolchain.

n8behavior committed:
Fix doc string formatting and test configuration

- Trim leading spaces from doc comment strings (standard Rust convention)
- Update test-book configuration to use correct relative paths
- Now test output matches expected output perfectly

Tests passing with correct mdbook preprocessing.

n8behavior committed:
Add HTML output to test book configuration

mythmon commented on .gitignore:
```diff
@@ -1 +1,2 @@
 /target
+.claude/settings.local.json
```
I'd prefer to keep personal tools preferences out of the general gitignore. Can you remove this?

n8behavior replied:
Ah, yes. Missed that. I'll get that removed

mythmon reviewed (changes requested):
Wow, I'd forgotten about this project. Thank you for the updates. There is one small charge that I'd like to see to this branch, but besides that I'd be happy to merge this.

n8behavior committed:
feat: add support for function documentation extraction

- Add support for Item::Fn in find_attrs_in_item
- Allows extracting documentation from standalone functions
- Functions can now be referenced with {{ #rustdoc crate::function_name }}

n8behavior commented:
Looks like I should remove the HTML format as well...

n8behavior committed:
revert: restore test-book/book.toml to upstream version

- Remove unnecessary path changes that could break CI/CD
- Keep test configuration identical to upstream

n8behavior committed:
revert: restore .gitignore to upstream version

- Remove tool-specific entries (.claude)
- Keep only essential /target ignore

mythmon commented:
Thanks for the updates!

n8behavior commented:
> Thanks for the updates!

Thank you! I was about to code this up from scratch. This was exactly what I was looking for.  My team and I love not having to duplicate Rustdoc comments into mdbook. Don't be surprised if you see some more PRs in the future.
